File: Baekjoon/노수일/Q1932.py
n = int(input())

triangle = []

for i in range(n):
    line_list = [0] * n
    input_lines = [int(num) for num in input().split(' ')]
    line_list[:len(input_lines)] = input_lines
    triangle.append(line_list)

"""
7 
3 8 
8 1 0 
2 7 4 4
4 5 2 6 5
"""

for i in range(1, n):
    for j in range(i + 1):
        if j == 0:
            triangle[i][j] = triangle[i][j] + triangle[i - 1][j]
        elif i == j:
            triangle[i][j] = triangle[i][j] + triangle[i - 1][j - 1]
        else:
            triangle[i][j] = triangle[i][j] + max(triangle[i - 1][j], triangle[i - 1][j - 1])

print(max(triangle[n - 1]))

File: Baekjoon/유석환/Q1932.swift
let n = Int(readLine()!)!
var triangle = [[Int]]()

triangle.append([Int(readLine()!)!])
for i in 1..<n {
    var input = readLine()!.split(separator: " ").map { Int(String($0))! }
    
    input[0] += triangle[i - 1][0]
    for j in 1..<i {
        input[j] += max(triangle[i - 1][j - 1], triangle[i - 1][j])
    }
    input[i] += triangle[i - 1][i - 1]
    
    triangle.append(input)
}

print(triangle[n - 1].max()!)
